safe/reportv4/test/test_impact_report.py

```diff
@@ -0,0 +1,304 @@
+# coding=utf-8
+import io
+import os
+import unittest
+from jinja2.environment import Template
+
+from safe.common.utilities import safe_dir
+from safe.gui.tools.minimum_needs.needs_profile import NeedsProfile
+from safe.reportv4.extractors.action_notes import (
+    action_checklist_extractor,
+    notes_assumptions_extractor)
+from safe.reportv4.extractors.analysis_detail import analysis_detail_extractor
+from safe.reportv4.extractors.analysis_result import analysis_result_extractor
+from safe.reportv4.extractors.composer import qgis_composer_extractor
+from safe.reportv4.extractors.impact_table import impact_table_extractor
+from safe.reportv4.extractors.minimum_needs import minimum_needs_extractor
+from safe.reportv4.processors.default import (
+    qgis_composer_renderer,
+    jinja2_renderer)
+from safe.reportv4.report_metadata import ReportMetadata
+from safe.test.utilities import get_qgis_app
+from safe.utilities.keyword_io import KeywordIO
+
+QGIS_APP, CANVAS, IFACE, PARENT = get_qgis_app()
+
+from PyQt4.QtCore import QSettings
+from qgis.core import QgsVectorLayer, QgsMapLayerRegistry
+from safe.definitionsv4.report import report_a3_portrait_blue, \
+    standard_impact_report_metadata
+from safe.reportv4.impact_report import ImpactReport
+
+__copyright__ = "Copyright 2016, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = ':%H$'
+
+
+class TestImpactReport(unittest.TestCase):
+
+    @classmethod
+    def fixtures_dir(cls, path):
+        dirname = os.path.dirname(__file__)
+        return os.path.join(dirname, 'fixtures', path)
+
+    def assertCompareFileControl(self, control_path, actual_path):
+        current_directory = safe_dir(sub_dir='../resources')
+        context = {
+            'current_directory': current_directory
+        }
+        with open(control_path) as control_file:
+            template_string = control_file.read()
+            template = Template(template_string)
+            control_string = template.render(context).strip()
+
+        with io.open(actual_path, encoding='utf-8') as actual_file:
+            actual_string = actual_file.read().strip()
+            self.assertEquals(control_string, actual_string)
+
+    @classmethod
+    def patch_keywords(cls, layer):
```

Comment:
I searched this but I didn't found it the first time, so I give up :v.
Anyway, I will replace this.